frontend/src/components/widget/CostPrice.js

```diff
@@ -0,0 +1,60 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { createAmount } from '../../utils/tableHelpers';
+export default class CostPrice extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.costRef = React.createRef();
+    this.state = { editMode: false };
+  }
+
+  /**
+   * @method localOnBlur
+   * @summary - proxy onBlur that also sets the visibility before passing further the event to the onBlur function from the props
+   * @param {DOM event} e
+   */
+  localOnBlur = (e) => {
+    const { onBlur } = this.props;
+    this.setState({ editMode: false });
+    onBlur(e);
+  };
+
+  render() {
+    const { value, precision } = this.props;
+    const { editMode } = this.state;
+
+    return (
+      <div>
+        {!editMode && (
+          <input
+            className="input-field js-input-field"
+            value={createAmount(value, precision)}
+            type="text"
+            onChange={() => false}
+            onFocus={() => {
+              this.setState({ editMode: true }, () => {
+                this.costRef.current.focus();
+              });
+            }}
+          />
+        )}
+
+        {editMode && (
+          <input
+            {...this.props}
+            onBlur={(e) => this.localOnBlur(e)}
```

Comment:
You can remove the anonymous function and just do `onBlur={this.localOnBlur}`

Reply:
Done. Thanks